zanata-war/src/main/java/org/zanata/dao/RequestDAO.java

```diff
@@ -0,0 +1,53 @@
+package org.zanata.dao;
+
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.jboss.seam.ScopeType;
+import org.jboss.seam.annotations.AutoCreate;
+import org.jboss.seam.annotations.Name;
+import org.jboss.seam.annotations.Scope;
+import org.zanata.model.LanguageRequest;
+import org.zanata.model.Request;
+
+import java.util.List;
+
+/**
+ * @author Alex Eng <a href="aeng@redhat.com">aeng@redhat.com</a>
+ */
+@Name("requestDAO")
+@AutoCreate
+@Scope(ScopeType.STATELESS)
+public class RequestDAO extends AbstractDAOImpl<Request, Long> {
+
+    public RequestDAO() {
+        super(Request.class);
+    }
+
+    public RequestDAO(Session session) {
+        super(Request.class, session);
+    }
+
+    public Request getById(Long requestId) {
+        return findById(requestId);
+    }
+
+    public List<Request> getByEntityId(String entityId) {
+        String query =
+            "from Request req where req.entityId= :entityId order by req.validFrom";
+        Query q = getSession().createQuery(query)
+            .setParameter("entityId", entityId)
+            .setCacheable(true).setComment(
+                "requestDAO.getByEntityId");
+        return q.list();
+    }
+
+    public Request getPendingRequestByEntityId(String entityId) {
```

Comment:
I think the most common case for these time-based entities would be "get me the latest version of entity", so I think it would be better to name this one "getByEntityId", and name the one which returns a list "getHistoryByEntityId".

Also, perhaps these methods could be declared in an AbstractTimeBasedDAO to help ensure consistency - even if all the methods are abstract for now.